oneandone/src/test/java/org/apache/jclouds/oneandone/rest/features/FirewallPolicyApiLiveTest.java

```diff
@@ -0,0 +1,188 @@
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+@Test(groups = "live", testName = "FirewallPolicyApiLiveTest")
+public class FirewallPolicyApiLiveTest extends BaseOneAndOneLiveTest {
+
+   private FirewallPolicy currentFirewallPolicy;
+   private Server currentServer;
+   private List<FirewallPolicy> firewallPolicies;
+
+   private FirewallPolicyApi firewallPolicyApi() {
+
+      return api.firewallPolicyApi();
+   }
+
+   @BeforeClass
+   public void setupTest() {
+      currentServer = createServer("firewallpolicies jclouds server");
+      assertNodeAvailable(currentServer);
+
+      List<FirewallPolicy.Rule.CreatePayload> rules = new ArrayList<FirewallPolicy.Rule.CreatePayload>();
+      FirewallPolicy.Rule.CreatePayload rule = FirewallPolicy.Rule.CreatePayload.builder()
+              .portFrom(80)
+              .portTo(80)
+              .protocol(Types.RuleProtocol.TCP)
+              .source("0.0.0.0")
+              .build();
+      rules.add(rule);
+      currentFirewallPolicy = firewallPolicyApi().create(FirewallPolicy.CreateFirewallPolicy.create("jclouds firewall policy", "desc", rules));
+   }
+
+   @Test
+   public void testList() {
+      firewallPolicies = firewallPolicyApi().list();
+
+      assertNotNull(firewallPolicies);
+      assertFalse(firewallPolicies.isEmpty());
+      Assert.assertTrue(firewallPolicies.size() > 0);
+   }
+
+   @Test
+   public void testListWithOption() {
+      GenericQueryOptions options = new GenericQueryOptions();
+      options.options(0, 0, null, "jclouds", null);
+      List<FirewallPolicy> imageWithQuery = firewallPolicyApi().list(options);
+
+      assertNotNull(imageWithQuery);
+      assertFalse(imageWithQuery.isEmpty());
+      Assert.assertTrue(imageWithQuery.size() > 0);
+   }
+
+   @Test
+   public void testGet() {
+      FirewallPolicy result = firewallPolicyApi().get(currentFirewallPolicy.id());
+
+      assertNotNull(result);
+      assertEquals(result.id(), currentFirewallPolicy.id());
+   }
+
+   @Test
+   public void testUpdate() {
+      String updatedName = "UpdatedjcloudsPolicy";
+
+      FirewallPolicy updateResult = firewallPolicyApi().update(currentFirewallPolicy.id(), FirewallPolicy.UpdateFirewallPolicy.create(updatedName, null));
+
+      assertNotNull(updateResult);
+      assertEquals(updateResult.name(), updatedName);
+   }
+
+   @Test(dependsOnMethods = "testUpdate")
+   public void testAssignServerIp() {
+      assertNodeAvailable(currentServer);
+      currentServer = updateServerStatus(currentServer);
+      List<String> ips = new ArrayList<String>();
+      ips.add(currentServer.ips().get(0).id());
+      FirewallPolicy.ServerIp.CreateServerIp toAdd = FirewallPolicy.ServerIp.CreateServerIp.create(ips);
+
+      FirewallPolicy updateResult = firewallPolicyApi().assignServerIp(currentFirewallPolicy.id(), toAdd);
+
+      assertNotNull(updateResult);
```

Comment:
Is there a more solid way to test that the result of the operation is the expected one?